### PR TITLE
Creates offstation variants of the protolathe and circuit imprinter for offstation and ghost roles.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Creates a more limited variant of the Protolathe and Circuit Imprinter for offstation roles.

The following limitations are applied to offstation Protolathes and Circuit Imprinters:
#### No general bluespace designs except Mining Satchel of Holding and Bluespace Crystals. This means no tracking beacons, bags of holding, GPS, desyncroniser or quantum spin inverter.

Free Golems are encouraged to spend their mining points on Marker Beacons to navigate. Everyone else is encouraged to build a cyborg and request it turn into the mining module for a more accessible GPS. This is experimental tech that should only be available to the machines on the futuristic research station.

GPS in general allows for the station to easily find ghost roles and for ghost roles to easily find the station. Ghost roles may be able to loot GPS off slain miners. Every space ruin has a GPS tucked away. A little bit of risk in exploring space can pay off dividends for ghost roles.

If the station wishes to integrate ghost roles into the station, the station can provide any missing techs from here to make this happen.

#### No Cargo Express Console upgrades or Supply Request Console related boards or NTnet Quantum Relay

These don't do anything anyway for ghost roles. Either require station access or station cooperation to set up. Useless to ghost roles alone. Station can provide them the tech if ghost roles really need it.

#### No crew monitors including handheld and computer boards.

Only available to the futuristic space station. Ghost roles have no reason to access crew monitors or wider crew info. No reason offstation ghost roles should be able to track crew.

#### No bluespace wormhole projectors.

Advanced Bluespace techs. Only available to the futuristic space station.

#### No teleportation station or teleportation hub boards including bluespace launchpads, quantum pads

Advanced Bluespace techs. Only available to the futuristic space station. Ghost roles now have to work harder to access the station z-level than building teleporters. This does not prevent the station bring teleporters to the ghost roles.

#### No Gigabeacons

Advanced Bluespace techs. No reason for ghost roles to advertise their presence. If the station is interesting in finding ghost roles, they can seek them out all the same.

## Why It's Good For The Game

There is little good reason for ghost roles to have an all-in-one unrestricted lathe that has magical access to all tech the station has researched. Certain high-impact designs can now be restricted from ghost roles if necessary.

## Changelog
:cl:
balance: Ghost role protolathes and circuit imprinters now have a more restricted design inventory, with fewer advanced or experimental designs relating to bluespace, quantum tech and teleportation.
/:cl:

### CODE REVIEW

**Comment by Livrah** [comment]
**No fun allowed**

**Comment by Livrah** [comment]
Good change, you could see free respawn golems on the station every cult round. What about miners vendor version of wormhole projector?

**Comment by Randomguy523** [comment]
Removal of GPS' in the fabs for being considered _'experimental tech'_ is downright silly, it's 2561, also would only add a small step for those who care enough to interact with the station.

As for the rest, removal of teleporters and the like is a good change, fairly recent technology that not everyone should have. 👍

**Comment by iwishforducks** [comment]
Could you keep the crew monitor and crew pinpointers, actually? I usually ask my fellow Charlie Crew to turn on their suit sensors so that I can track them down if they get injured or need assistance. It's especially important if you get an AI up, as they are unable to use PDA tracking to track us down. And doubly more so if GPSs are being removed.

Also, I do agree with removing teleporters, but I personally think quantum pads can stay.

**Comment by CRITAWAKETS** [comment]
they're literally on the same research server as the station, they're freebooting off the advanced bluespace research, not researching it themselves. Lore is also not the best excuse to tamper with balance. Also a GPS unit is not really an advanced technology.

that said i do agree with the removal of the tracking beacon, cargo consoles and (regular) teleporters, those are kinda impeding

File: code/modules/research/designs/AI_module_designs.dm [hunk 1]
```diff
@@ -113,7 +113,6 @@
 	name = "Core Module Design (P.A.L.A.D.I.N.)"
 	desc = "Allows for the construction of a P.A.L.A.D.I.N. AI Core Module."
 	id = "paladin_module"
-	build_type = IMPRINTER
```
Comment (CRITAWAKETS) [inline code/modules/research/designs/AI_module_designs.dm:116]: undoced removal of PALADIN from being buildable

Comment (CRITAWAKETS) [inline code/modules/research/designs/AI_module_designs.dm:116]: should probably document this in the PR

File: code/modules/research/designs/machine_designs.dm [hunk 2]
```diff
@@ -485,6 +511,7 @@
 	name = "Machine Design (NTNet Relay Board)"
 	desc = "The circuit board for a wireless network relay."
 	id = "ntnet_relay"
+	build_type = IMPRINTER
```
Comment (CRITAWAKETS) [inline code/modules/research/designs/machine_designs.dm:514]: Undoced balance change.

**Comment by Fikou** [comment]
if any golem wants to get to station, they can just travel south west, theres no reason to remove gps from them i think, especially if youre trying to do something fun like killing megafauna or getting back to your ship from a long travel. good change otherwise though

File: _maps/RandomZLevels/undergroundoutpost45.dmm [hunk 3]
```diff
@@ -3614,7 +3614,7 @@
 	},
 /area/awaymission/undergroundoutpost45/research)
 "hD" = (
-/obj/machinery/rnd/production/protolathe,
+/obj/machinery/rnd/production/protolathe/offstation,
```
Comment (Fikou) [inline _maps/RandomZLevels/undergroundoutpost45.dmm:3617]: no reason to bother here? its an away mission, accessible only from the gateway which is on the station, without any ghost roles i imagine

File: code/modules/research/designs/AI_module_designs.dm (hunk 1 above)
Comment (Timberpoes) [reply code/modules/research/designs/AI_module_designs.dm:116]: The change had no impact.

`/datum/design/board` parent type had `build_type = IMPRINTER` - This line was just re-defining it to be the same thing. If I didn't remove this, offstation imprinters would not be able to print them and there was no point in re-defining it as `build_type = IMPRINTER | AWAY_IMPRINTER` when `/datum/design/board` is now that.

tl;dr - Re-defined parent build_type and would have caused a bug if I left it in. Removing it has no functionality change beyond what the PR body already states.

File: _maps/RandomZLevels/undergroundoutpost45.dmm (hunk 3 above)
Comment (Timberpoes) [reply _maps/RandomZLevels/undergroundoutpost45.dmm:3617]: I just did a find-and-replace on all protolathes, since no non-departmental variants were onstation. But I can change it back, since you're right that it is just an away mission.

File: code/modules/research/designs/machine_designs.dm (hunk 2 above)
Comment (Timberpoes) [reply code/modules/research/designs/machine_designs.dm:514]: This could only be printed from the IMPRINTER before. Same as above. 

image
image

Although NTnet is just used for door remotes and the mulebot (?) and not teleportation as I originally thought.

There's also nothing to be gained by including more chaff that no ghost roles can actually use. I did the same when I trimmed the cargo consoles out. Not because they're something ghost roles shouldn't have, but because they're things ghost roles can't use. I'll move it to the appropriate category in the PR body. Unless you're telling me there's some massive use-case for ghost roles to build the quantum relay?

**Comment by Timberpoes** [comment]
The new imprinter and lathe variants are intended to specially be trimmed down versions. Ideally they should contain a set of useful core designs from each of the departmental lathes that are tailored specifically to ghost roles and follow the broad idea that ghost roles should not be abandoning where they are to go on flights of fancy.

They're not made with any sort of overarching goal in mind. The alternative is making a unique "department" for each space ruin and creating a special, very limited lathe specifically targetted at what that ghost role can expect to do.

> 
> 
> Removal of GPS' in the fabs for being considered _'experimental tech'_ is downright silly, it's 2561, also would only add a small step for those who care enough to interact with the station.
> 
> As for the rest, removal of teleporters and the like is a good change, fairly recent technology that not everyone should have. 👍

Quoting myself:
GPS in general allows for the station to easily find ghost roles and for ghost roles to easily find the station. Ghost roles may be able to loot GPS off slain miners. Every space ruin has a GPS tucked away. A little bit of risk in exploring space can pay off dividends for ghost roles.

Nothing here about them being experimental tech. Purely a balance tweak.

File: _maps/RandomZLevels/undergroundoutpost45.dmm (hunk 3 above)
Comment (Timberpoes) [reply _maps/RandomZLevels/undergroundoutpost45.dmm]: Restored.

File: code/modules/research/designs.dm [hunk 4]
```diff
@@ -27,7 +27,7 @@ other types of metals and chemistry for reagents).
 	var/desc = "Desc"
 	/// The ID of the design. Used for quick reference. Alphanumeric, lower-case, no symbols
 	var/id = DESIGN_ID_IGNORE
-	/// Bitflags indicating what machines this design is compatable with. ([IMPRINTER]|[PROTOLATHE]|[AUTOLATHE]|[MECHFAB]|[BIOGENERATOR]|[LIMBGROWER]|[SMELTER]|[NANITE_COMPILER])
+	/// Bitflags indicating what machines this design is compatable with. ([IMPRINTER]|[AWAY_IMPRINTER]|[PROTOLATHE]|[AWAY_LATHE]|[AUTOLATHE]|[MECHFAB]|[BIOGENERATOR]|[LIMBGROWER]|[SMELTER]|[NANITE_COMPILER])
```
Comment (Mothblocks) [inline code/modules/research/designs.dm:30]: This should really just reference a defines file so we don't have to keep updating it. Oh well.

**Comment by adamkad1** [comment]
Why is this trash merged

**Comment by Mothblocks** [comment]
sir this pr is from march
